pkg/scaffold/internal/filesystem/errors.go

```diff
@@ -24,6 +24,38 @@ import (
 // They are not exported as they should not be created outside of this package
 // Exported functions are provided to check which kind of error was returned
 
+// fileExistsError is returned if it could not be checked if the file exists
+type fileExistsError struct {
+	path string
+	err  error
+}
+
+func (e fileExistsError) Error() string {
+	return fmt.Sprintf("failed to check if %s exists: %v", e.path, e.err)
+}
+
+// IsFileExistsError checks if the returned error is because the file could not be checked for existence
+func IsFileExistsError(e error) bool {
```

Comment:
we should eventually switch these all to use `errors.As`

Reply:
I agree, I submited a PR for that and didnt get any answer, it can be done after this gets merged.